### PR TITLE
chore(j-s): Court API Availability

### PR DESCRIPTION
**Pull request by gudjong**
# Court API Availability

[Ekki tala við Auði í umhverfum þar sem hún er ekki aðgengileg](https://app.asana.com/0/1199153462262248/1202882386999270/f)

## What

- Avoid calling the court API in environments where it is not available.

## Why

We want to prevent unnecessary delays when responding to user requests and we want to avoid spamming error logs.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by oddsson — APPROVED** [review]

File: apps/judicial-system/backend/src/app/modules/court/court.service.ts [hunk 1]
```diff
@@ -117,12 +129,15 @@ export class CourtService {
     fileName: string,
     content: Buffer,
   ): Promise<string> {
-    return this.uploadStream(
-      courtId,
-      `${fileName}.pdf`,
-      'application/pdf',
-      content,
-    )
+    return this.confirmApiAvailability()
```
Comment (Arnarkari93) [inline apps/judicial-system/backend/src/app/modules/court/court.service.ts:132]: I feel like there should be a safer way of doing this. One could easily forget to call this method first before trying to access `this.courtClientService`. When this config is set, could we inject a striped-out implementation of the `CourtClientService`?

**Review by Arnarkari93 — CHANGES_REQUESTED** [review]
One thought.

File: apps/judicial-system/backend/src/app/modules/court/court.service.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/court/court.service.ts:132]: You are right of course, but I am not sure it is worth the effort. I use the same approach in the police service when calling LÖKE. I'm going to ponder this a bit and see if I can easily come up with a more elegant solution.

Comment (Arnarkari93) [reply apps/judicial-system/backend/src/app/modules/court/court.service.ts:132]: I see, I will take the auto merge label off and accept. Then you can let this through if you want.

**Review by Arnarkari93 — APPROVED** [review]

**Review by Arnarkari93 — APPROVED** [review]

**Pull request merged**